File: petstore/typescript/src/models/pet.ts
import {createCategoryFromDiscriminatorValue} from './createCategoryFromDiscriminatorValue';
import {createTagFromDiscriminatorValue} from './createTagFromDiscriminatorValue';
import {Category, Tag} from './index';
import {Pet_status} from './pet_status';
import {AdditionalDataHolder, Parsable, ParseNode, SerializationWriter} from '@microsoft/kiota-abstractions';

export class Pet implements AdditionalDataHolder, Parsable {
    /** Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well. */
    private _additionalData: Record<string, unknown>;
    /** The category property */
    private _category?: Category | undefined;
    /** The id property */
    private _id?: number | undefined;
    /** The name property */
    private _name?: string | undefined;
    /** The photoUrls property */
    private _photoUrls?: string[] | undefined;
    /** pet status in the store */
    private _status?: Pet_status | undefined;
    /** The tags property */
    private _tags?: Tag[] | undefined;
    /**
     * Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @returns a Record<string, unknown>
     */
    public get additionalData() {
        return this._additionalData;
    };
    /**
     * Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param value Value to set for the AdditionalData property.
     */
    public set additionalData(value: Record<string, unknown>) {
        this._additionalData = value;
    };
    /**
     * Gets the category property value. The category property
     * @returns a Category
     */
    public get category() {
        return this._category;
    };
    /**
     * Sets the category property value. The category property
     * @param value Value to set for the category property.
     */
    public set category(value: Category | undefined) {
        this._category = value;
    };
    /**
     * Instantiates a new Pet and sets the default values.
     */
    public constructor() {
        this._additionalData = {};
    };
    /**
     * The deserialization information for the current model
     * @returns a Record<string, (node: ParseNode) => void>
     */
    public getFieldDeserializers() : Record<string, (node: ParseNode) => void> {
        return {
            "category": n => { this.category = n.getObjectValue<Category>(createCategoryFromDiscriminatorValue); },
            "id": n => { this.id = n.getNumberValue(); },
            "name": n => { this.name = n.getStringValue(); },
            "photoUrls": n => { this.photoUrls = n.getCollectionOfPrimitiveValues<string>(); },
            "status": n => { this.status = n.getEnumValue<Pet_status>(Pet_status); },
            "tags": n => { this.tags = n.getCollectionOfObjectValues<Tag>(createTagFromDiscriminatorValue); },
        };
    };
    /**
     * Gets the id property value. The id property
     * @returns a int64
     */
    public get id() {
        return this._id;
    };
    /**
     * Sets the id property value. The id property
     * @param value Value to set for the id property.
     */
    public set id(value: number | undefined) {
        this._id = value;
    };
    /**
     * Gets the name property value. The name property
     * @returns a string
     */
    public get name() {
        return this._name;
    };
    /**
     * Sets the name property value. The name property
     * @param value Value to set for the name property.
     */
    public set name(value: string | undefined) {
        this._name = value;
    };
    /**
     * Gets the photoUrls property value. The photoUrls property
     * @returns a string
     */
    public get photoUrls() {
        return this._photoUrls;
    };
    /**
     * Sets the photoUrls property value. The photoUrls property
     * @param value Value to set for the photoUrls property.
     */
    public set photoUrls(value: string[] | undefined) {
        this._photoUrls = value;
    };
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    public serialize(writer: SerializationWriter) : void {
        if(!writer) throw new Error("writer cannot be undefined");
        writer.writeObjectValue<Category>("category", this.category);
        writer.writeNumberValue("id", this.id);
        writer.writeStringValue("name", this.name);
        writer.writeCollectionOfPrimitiveValues<string>("photoUrls", this.photoUrls);
        writer.writeEnumValue<Pet_status>("status", this.status);
        writer.writeCollectionOfObjectValues<Tag>("tags", this.tags);
        writer.writeAdditionalData(this.additionalData);
    };
    /**
     * Gets the status property value. pet status in the store
     * @returns a Pet_status
     */
    public get status() {
        return this._status;
    };
    /**
     * Sets the status property value. pet status in the store
     * @param value Value to set for the status property.
     */
    public set status(value: Pet_status | undefined) {
        this._status = value;
    };
    /**
     * Gets the tags property value. The tags property
     * @returns a Tag
     */
    public get tags() {
        return this._tags;
    };
    /**
     * Sets the tags property value. The tags property
     * @param value Value to set for the tags property.
     */
    public set tags(value: Tag[] | undefined) {
        this._tags = value;
    };
}

File: petstore/go/utilities/models/pet.go
package models

import (
    i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91 "github.com/microsoft/kiota-abstractions-go/serialization"
)

type Pet struct {
    // Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
    additionalData map[string]any
    // The category property
    category Categoryable
    // The id property
    id *int64
    // The name property
    name *string
    // The photoUrls property
    photoUrls []string
    // pet status in the store
    status *Pet_status
    // The tags property
    tags []Tagable
}
// NewPet instantiates a new Pet and sets the default values.
func NewPet()(*Pet) {
    m := &Pet{
    }
    m.SetAdditionalData(make(map[string]any))
    return m
}
// CreatePetFromDiscriminatorValue creates a new instance of the appropriate class based on discriminator value
// returns a Parsable when successful
func CreatePetFromDiscriminatorValue(parseNode i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode)(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.Parsable, error) {
    return NewPet(), nil
}
// GetAdditionalData gets the AdditionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
// returns a map[string]any when successful
func (m *Pet) GetAdditionalData()(map[string]any) {
    return m.additionalData
}
// GetCategory gets the category property value. The category property
// returns a Categoryable when successful
func (m *Pet) GetCategory()(Categoryable) {
    return m.category
}
// GetFieldDeserializers the deserialization information for the current model
// returns a map[string]func(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode)(error) when successful
func (m *Pet) GetFieldDeserializers()(map[string]func(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode)(error)) {
    res := make(map[string]func(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode)(error))
    res["category"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        val, err := n.GetObjectValue(CreateCategoryFromDiscriminatorValue)
        if err != nil {
            return err
        }
        if val != nil {
            m.SetCategory(val.(Categoryable))
        }
        return nil
    }
    res["id"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        val, err := n.GetInt64Value()
        if err != nil {
            return err
        }
        if val != nil {
            m.SetId(val)
        }
        return nil
    }
    res["name"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        val, err := n.GetStringValue()
        if err != nil {
            return err
        }
        if val != nil {
            m.SetName(val)
        }
        return nil
    }
    res["photoUrls"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        val, err := n.GetCollectionOfPrimitiveValues("string")
        if err != nil {
            return err
        }
        if val != nil {
            res := make([]string, len(val))
            for i, v := range val {
                if v != nil {
                    res[i] = *(v.(*string))
                }
            }
            m.SetPhotoUrls(res)
        }
        return nil
    }
    res["status"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        val, err := n.GetEnumValue(ParsePet_status)
        if err != nil {
            return err
        }
        if val != nil {
            m.SetStatus(val.(*Pet_status))
        }
        return nil
    }
    res["tags"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        val, err := n.GetCollectionOfObjectValues(CreateTagFromDiscriminatorValue)
        if err != nil {
            return err
        }
        if val != nil {
            res := make([]Tagable, len(val))
            for i, v := range val {
                if v != nil {
                    res[i] = v.(Tagable)
                }
            }
            m.SetTags(res)
        }
        return nil
    }
    return res
}
// GetId gets the id property value. The id property
// returns a *int64 when successful
func (m *Pet) GetId()(*int64) {
    return m.id
}
// GetName gets the name property value. The name property
// returns a *string when successful
func (m *Pet) GetName()(*string) {
    return m.name
}
// GetPhotoUrls gets the photoUrls property value. The photoUrls property
// returns a []string when successful
func (m *Pet) GetPhotoUrls()([]string) {
    return m.photoUrls
}
// GetStatus gets the status property value. pet status in the store
// returns a *Pet_status when successful
func (m *Pet) GetStatus()(*Pet_status) {
    return m.status
}
// GetTags gets the tags property value. The tags property
// returns a []Tagable when successful
func (m *Pet) GetTags()([]Tagable) {
    return m.tags
}
// Serialize serializes information the current object
func (m *Pet) Serialize(writer i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.SerializationWriter)(error) {
    {
        err := writer.WriteObjectValue("category", m.GetCategory())
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteInt64Value("id", m.GetId())
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteStringValue("name", m.GetName())
        if err != nil {
            return err
        }
    }
    if m.GetPhotoUrls() != nil {
        err := writer.WriteCollectionOfStringValues("photoUrls", m.GetPhotoUrls())
        if err != nil {
            return err
        }
    }
    if m.GetStatus() != nil {
        cast := (*m.GetStatus()).String()
        err := writer.WriteStringValue("status", &cast)
        if err != nil {
            return err
        }
    }
    if m.GetTags() != nil {
        cast := make([]i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.Parsable, len(m.GetTags()))
        for i, v := range m.GetTags() {
            if v != nil {
                cast[i] = v.(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.Parsable)
            }
        }
        err := writer.WriteCollectionOfObjectValues("tags", cast)
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteAdditionalData(m.GetAdditionalData())
        if err != nil {
            return err
        }
    }
    return nil
}
// SetAdditionalData sets the AdditionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
func (m *Pet) SetAdditionalData(value map[string]any)() {
    m.additionalData = value
}
// SetCategory sets the category property value. The category property
func (m *Pet) SetCategory(value Categoryable)() {
    m.category = value
}
// SetId sets the id property value. The id property
func (m *Pet) SetId(value *int64)() {
    m.id = value
}
// SetName sets the name property value. The name property
func (m *Pet) SetName(value *string)() {
    m.name = value
}
// SetPhotoUrls sets the photoUrls property value. The photoUrls property
func (m *Pet) SetPhotoUrls(value []string)() {
    m.photoUrls = value
}
// SetStatus sets the status property value. pet status in the store
func (m *Pet) SetStatus(value *Pet_status)() {
    m.status = value
}
// SetTags sets the tags property value. The tags property
func (m *Pet) SetTags(value []Tagable)() {
    m.tags = value
}
type Petable interface {
    i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.AdditionalDataHolder
    i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.Parsable
    GetCategory()(Categoryable)
    GetId()(*int64)
    GetName()(*string)
    GetPhotoUrls()([]string)
    GetStatus()(*Pet_status)
    GetTags()([]Tagable)
    SetCategory(value Categoryable)()
    SetId(value *int64)()
    SetName(value *string)()
    SetPhotoUrls(value []string)()
    SetStatus(value *Pet_status)()
    SetTags(value []Tagable)()
}
